core/src/main/kotlin/org/evomaster/core/database/SqlInsertBuilder.kt

```diff
@@ -121,6 +129,78 @@ class SqlInsertBuilder(
         }
     }
 
+    private fun findRangeConstraint(tableConstraints: List<TableConstraint>, columnName: String): RangeConstraint? {
+        return tableConstraints.filter { c -> c is RangeConstraint }
+                .map { c -> c as RangeConstraint }
+                .firstOrNull { c -> c.columnName.equals(columnName, true) }
+    }
+
+    private fun findLowerBound(tableConstraints: List<TableConstraint>, columnName: String): Int? {
+        val rangeConstraint = findRangeConstraint(tableConstraints, columnName)
+        if (rangeConstraint != null) {
+            return rangeConstraint.minValue.toInt()
+        }
+
+        val lowerBounds = tableConstraints
+                .asSequence()
+                .filter { c -> c is LowerBoundConstraint }
+                .map { c -> c as LowerBoundConstraint }
+                .filter { c -> c.columnName.equals(columnName, true) }
+                .map { c -> c.lowerBound.toInt() }
+                .toList()
```

Comment:
Kotlin (in contrast to Java) has a stream function called `filterIsInstance` which could simplify the code here, as it does a `map` as well in a single call. Could apply also in the other places in this file where there is similar code

Reply:
Thanks. I changed three parts where the filter{...}.map{...} was being used with a corresponding filterIsInstance().